packages/edit-site/src/components/style-book/index.js

```diff
@@ -156,65 +241,105 @@ function StyleBook( { isSelected, onSelect, onClose } ) {
+											// @see https://developer.mozilla.org/en-US/docs/Web/Guide/CSS/Block_formatting_context
+											`.is-root-container { display: flow-root; }
+											body { position: relative; padding: 32px !important; }` +
+												STYLE_BOOK_IFRAME_STYLES
+										}
+									</style>
+								</>
+							}
+							name="style-book-canvas"
+							tabIndex={ 0 }
+						>
+							{ /* Filters need to be rendered before children to avoid Safari rendering issues. */ }
+							{ settings.svgFilters }
+							<Examples
+								className={ classnames(
+									'edit-site-style-book__examples',
+									{
+										'is-wide': sizes.width > 600,
+									}
+								) }
+								examples={ examples }
+								category={ tab.name }
+								isSelected={ isSelected }
+								onSelect={ onSelect }
+							/>
+						</Iframe>
 					) }
 				</TabPanel>
 			</section>
 		</StyleBookFill>
 	);
 }
 
-const Examples = memo( ( { examples, category, isSelected, onSelect } ) => (
-	<div className="edit-site-style-book__examples">
-		{ examples
-			.filter( ( example ) => example.category === category )
-			.map( ( example ) => (
-				<Example
-					key={ example.name }
-					title={ example.title }
-					blocks={ example.blocks }
-					isSelected={ isSelected( example.name ) }
-					onClick={ () => {
-						onSelect( example.name );
-					} }
-				/>
-			) ) }
-	</div>
-) );
-
-const Example = memo( ( { title, blocks, isSelected, onClick } ) => (
-	<button
-		className={ classnames( 'edit-site-style-book__example', {
-			'is-selected': isSelected,
-		} ) }
-		aria-label={ sprintf(
-			// translators: %s: Title of a block, e.g. Heading.
-			__( 'Open %s styles in Styles panel' ),
-			title
-		) }
-		onClick={ onClick }
-	>
-		<span className="edit-site-style-book__example-title">{ title }</span>
-		<div className="edit-site-style-book__example-preview">
-			<BlockPreview
-				blocks={ blocks }
-				viewportWidth={ 0 }
-				additionalStyles={ [
-					{
-						css:
-							'.wp-block:first-child { margin-top: 0; }' +
-							'.wp-block:last-child { margin-bottom: 0; }',
-					},
-				] }
-			/>
+const Examples = memo(
+	( { className, examples, category, isSelected, onSelect } ) => (
+		<div className={ className }>
+			{ examples
+				.filter( ( example ) => example.category === category )
+				.map( ( example ) => (
+					<Example
+						key={ example.name }
+						title={ example.title }
+						blocks={ example.blocks }
+						isSelected={ isSelected( example.name ) }
+						onClick={ () => {
+							onSelect( example.name );
+						} }
+					/>
+				) ) }
 		</div>
-	</button>
-) );
+	)
+);
+
+const Example = memo( ( { title, blocks, isSelected, onClick } ) => {
+	const originalSettings = useSelect(
+		( select ) => select( blockEditorStore ).getSettings(),
+		[]
+	);
+	const settings = useMemo(
+		() => ( { ...originalSettings, __unstableIsPreviewMode: true } ),
+		[ originalSettings ]
+	);
+
+	return (
+		<button
```

Comment:
This can lead to invalid HTML(`<button> cannot appear as a descendant of <button>`).  Can we avoid this by using `Composite`?

Reply:
That's a good point, I don't think we have any actual buttons in core blocks, but we do have other interactive elements. 

I wonder if we could wrap the whole thing in a div, and use a button to wrap the block name, and then have the actual block sit beside it. It might be trickier to activate the outline when the button is clicked but we may be able to leverage `:has()` for that?

Reply:
Thanks for the feedback! I'll have a go at using `Composite`, I see the [block patterns list uses `Composite`](https://github.com/WordPress/gutenberg/blob/94ff2ba55379d9ad7f6bed743b20b85ff26cf56d/packages/block-editor/src/components/block-patterns-list/index.js#L130) and `CompositeItem` so it looks like we'll have some consistency there if we switch to that.

>  I don't think we have any actual buttons in core blocks, but we do have other interactive elements.

I tried installing a couple of plugins and I noticed that WooCommerce includes buttons in their cart block, so I imagine it's fairly likely that we'll encounter more 3rd party blocks that use `button` elements.

I'll have a play and see what our options are 🙂

Reply:
Note that whatever we go with here will still be an improvement over trunk, where we have a whole iframe inside each button 😅

Reply:
I've had a go with Composite and CompositeItem in https://github.com/WordPress/gutenberg/pull/48664/commits/cf0aec6de0afc5c7914f4d72a32e564e5387a6cc, seems to be working pretty well so far!